### PR TITLE
commands/service:doc: save generated markdown to readme file

### PR DESCRIPTION
**Pull request by ilgooz**

### CODE REVIEW

**Comment by antho1404** [comment]
I was thinking of keeping it in the console and just use `>> readme.md` or whatever file like that it's more flexible

**Comment by NicolasMahe** [comment]
I prefer @ilgooz's version to directly write in a file. The target are not so expert dev.

**Comment by antho1404** [comment]
I could be ok with that, just with file we will have big limitations like cannot just happen in a file or also that we enforce only one possible name for the readme that if we let it free we don't have to take such decisions and if users don't like to use `>>` they can copy/paste.

I don't want to spend too much time on that and I'm fine on having this in a file just make sure that we all see the limitations and that we are taking a strong decision that maybe developers will not like where with a basic output in the stdout, we are not taking any decisions and the developer have the choice.

**Comment by ilgooz** [comment]
how about adding a `--save` option? when used empty, we can replace the readme and when used with a file name we can write to that file. if not provided we can just dump to standard output.

**Comment by antho1404** [comment]
this is already way better, not sure this is required to support right now but the PR is already here and this modification small so i would be ok with that

**Comment by ilgooz** [comment]
> this is already way better, not sure this is required to support right now but the PR is already here and this modification small so i would be ok with that

In that case, I'm adding the `--save` option in this PR. Is this what you agree on, can you please clarify?

**Comment by antho1404** [comment]
> I'm adding the --save option in this PR

agree

File: src/commands/service/doc.ts [hunk 1]
```diff
@@ -12,6 +13,7 @@ export default class ServiceDoc extends Command {
 
   static flags = {
     ...Command.flags,
+    save: flags.string({char: 's', description: 'Save to specified file or allow to overwrite the default readme file when used as empty or create a new one if not exists'}),
```
Comment (antho1404) [inline src/commands/service/doc.ts]: hum in fact the ux is weird with `--save file` maybe it should be one of these:
- `--save --filename xxx`
- `--output file`

What do you think?

Comment (ilgooz) [reply src/commands/service/doc.ts]: 1st opt is ok with me. I don't know why cmd lib requires a following `=` for empty values on flags.

Comment (NicolasMahe) [reply src/commands/service/doc.ts]: why not simply `--save` without asking the user to set the path to the file?
Let's keep it simple. If the guy want to output the content in another file, then he can copy / paste or use >>

Comment (NicolasMahe) [reply src/commands/service/doc.ts]: From the example, if the flag is of type bool, then user doesn't have to set a value or =
https://oclif.io/docs/flags

File: src/commands/service/doc.ts [hunk 2]
```diff
@@ -33,4 +35,23 @@ export default class ServiceDoc extends Command {
...
+    let readmeFileName = saveFileName || 'README.md'
+    if (!saveFileName) {
+      let defaultReadmeFileName = readdirSync(servicePath).find(file => {
+        return /^readme(?:.(?:md|txt)+)?$/i.test(file)
+      })
+      if (defaultReadmeFileName) {
+        readmeFileName = defaultReadmeFileName
+      }
+    }
+    writeFile(join(servicePath, readmeFileName), markdown, (err) => {
```
Comment (NicolasMahe) [inline src/commands/service/doc.ts]: should use sync version of `writeFile` function or return a primise.
The callback is throwing an err that nothing will catch..

File: src/commands/service/doc.ts (hunk 1 above)
Comment (antho1404) [reply src/commands/service/doc.ts]: agree, KISS

File: src/commands/service/doc.ts (hunk 2 above)
Comment (ilgooz) [reply src/commands/service/doc.ts]: done

File: src/commands/service/doc.ts (hunk 1 above)
Comment (ilgooz) [reply src/commands/service/doc.ts]: done

**Pull request merged**